lib/msgServer/mmrp/index.js

```diff
@@ -168,25 +221,29 @@ MmrpNode.prototype.connect = function (uri, clusterId, cb) {
  * Disconnects the dealer from a given URI (if connected)
  *
  * @param {string} uri
+ * @param {string} clusterId
  */
 
-MmrpNode.prototype.disconnect = function (uri) {
-	var clusterIds = Object.keys(this.relays);
-	for (var i = 0; i < clusterIds.length; i += 1) {
-		var clusterId = clusterIds[i];
-		var conn = this.relays[clusterId];
-
-		if (conn.uri === uri) {
-			delete this.relays[clusterId];
+MmrpNode.prototype.disconnect = function (uri, clusterId) {
+	const relay = this.relays[clusterId];
+	const verbose = logger.verbose.data({
+		uri,
+		localClusterId: this.clusterId,
+		receivedClusterId: clusterId
+	});
 
-			logger.verbose(this.clusterId, 'dealer disconnecting from', uri, '(cluster: ' + clusterId + ')');
+	if (!relay) {
+		return verbose.log('dealer disconnect requested but relay does not exist');
+	}
 
-			this.dealer.disconnect(uri);
-			return;
-		}
+	if (uri !== relay.uri) {
```

Comment:
How could this ever happen?

Reply:
I'd say its something we'd want to have blow up in our face if it *did* ever happen.